db/migrate/617_create_customer_orders_view_version_4_up.sql

```diff
@@ -113,20 +122,21 @@ CREATE OR REPLACE VIEW unified_customer_orders AS
            STRING_AGG(COALESCE("is".inventory_code, ''), ' ') || ' ' ||
            STRING_AGG(COALESCE("is".serial_number, ''), ' ') ) AS searchable,
         'approved' = ANY(ARRAY_AGG(rs.status)) AS with_pickups,
-        'signed' = ANY(ARRAY_AGG(rs.status)) AS with_returns,
+        'signed' = ANY(ARRAY_AGG(rs.status)) OR 'signed' = ANY(ARRAY_AGG(rs2.status)) AS with_returns,
          co.created_at,
          co.updated_at,
          co.title,
          co.lending_terms_accepted,
          co.contact_details,
-         ARRAY_AGG(rs.id) AS reservation_ids,
-         ARRAY_AGG(DISTINCT rs.status) AS reservation_states,
+         ARRAY_UNIQUE_NO_NULLS(ARRAY_AGG(rs.id) || ARRAY_AGG(rs2.id)) AS reservation_ids,
+         ARRAY_UNIQUE_NO_NULLS(ARRAY_AGG(rs.status) || ARRAY_AGG(rs2.status)) AS reservation_states,
          'customer_orders' AS origin_table
   FROM customer_orders AS co
   JOIN orders AS os ON os.customer_order_id = co.id
   JOIN reservations AS rs ON rs.order_id = os.id
   JOIN models AS ms ON rs.model_id = ms.id
-  LEFT JOIN options AS ops ON rs.option_id = os.id
   LEFT JOIN items AS "is" ON rs.item_id = "is".id
   LEFT JOIN contracts AS cs ON rs.contract_id = cs.id
+  LEFT JOIN reservations AS rs2 ON rs2.contract_id = cs.id
+  LEFT JOIN options AS ops ON rs2.option_id = ops.id AND rs2.order_id IS NULL
```

Comment:
@urs-van-binsbergen I have moved this line, because it was a bug. Options can never be ordered.